samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/servlet/cp/DownloadCPServlet.java

```diff
@@ -79,34 +85,139 @@ public void doGet(HttpServletRequest req, HttpServletResponse res)
+						String manString = manifestGenerator.getManifest();
+						log.debug("manString = " + manString);
+						b = manString.getBytes();
+						zos.write(b, 0, b.length);
+						zos.closeEntry();
+
+						// Attachments
+						HashMap contentMap = manifestGenerator.getContentMap();
+
+						String filename = null;
+						for (Iterator it = contentMap.entrySet().iterator(); it.hasNext();) {
+							Map.Entry entry = (Map.Entry) it.next();
+							filename = (String)  entry.getKey();
+							ze = new ZipEntry(filename.substring(1));
+							zos.putNextEntry(ze);
+							b = (byte[]) entry.getValue();
+							zos.write(b, 0, b.length);
+							zos.closeEntry();
+						}
+
+					} catch (IOException e) {
+						log.error(e.getMessage());
+						throw e;
+					} finally {
+						if (zos != null) {
+							try {
+								zos.closeEntry();
+							} catch (IOException e) {
+								log.error(e.getMessage());
+							}
+							try {
+								zos.close();
+							} catch (IOException e) {
+								log.error(e.getMessage());
+							}
+
+						}
+					}
+				}
+			} else {
+				if (success == AssessmentService.UPDATE_ERROR_DRAW_SIZE_TOO_LARGE) {
+					String err=ContextUtil.getLocalizedString("org.sakaiproject.tool.assessment.bundle.AuthorMessages","update_pool_error_size_too_large");
+					req.setAttribute("error", err);
+				} else {
+					String err=ContextUtil.getLocalizedString("org.sakaiproject.tool.assessment.bundle.AuthorMessages","update_pool_error_unknown");
+					req.setAttribute("error", err);
+				}
+
+				String path = "/jsf/qti/poolUpdateError.faces";
+				RequestDispatcher dispatcher = req.getRequestDispatcher(path);
+				dispatcher.forward(req, res);
+			}
+		} else if (StringUtils.isNotEmpty(questionPoolId)) {
+			QuestionPoolService questionPoolService = new QuestionPoolService();
+			QuestionPoolFacade questionPool = questionPoolService.getPool(Long.parseLong(questionPoolId), AgentFacade.getAgentString());
+
+			List items = questionPoolService.getAllItems(questionPool.getQuestionPoolId());
+			StringBuilder sb = new StringBuilder();
+
+			// creating a question list separated by comma
+			for (Object item : items) {
+				if (sb.length() > 0) {
+					sb.append(",");
+				}
+				sb.append(((ItemFacade) item).getItemId());
 			}
 
+			// checking user can export pool
+			accessDenied = !questionPoolService.canExportPool(questionPoolId, agentIdString);
+
 			if (accessDenied) {
 				String path = "/jsf/qti/exportDenied.faces";
 				RequestDispatcher dispatcher = req.getRequestDispatcher(path);
 				dispatcher.forward(req, res);
 			} else {
 				res.setContentType("application/x-zip-compressed");
-				String zipFilename = "exportAssessment.zip";
-				res.setHeader("Content-Disposition", "attachment;filename=\""
-						+ zipFilename + "\";");
+				String zipFilename = "exportPool.zip";
+				res.setHeader("Content-Disposition", "attachment;filename=\"" + zipFilename + "\";");
```

Comment:
Same, constants are great, this is an example of commons-fileuploads

https://commons.apache.org/proper/commons-fileupload/apidocs/constant-values.html